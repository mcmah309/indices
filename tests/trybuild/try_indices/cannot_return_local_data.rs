use indices::try_indices;

fn main() {
    one();
    two();
    three();
    four();
    five();
}

fn one() -> Result<&'static mut i32, indices::TryIndicesError> {
    let mut data = [5, 4, 3, 2, 1];
    let slice = data.as_mut_slice();
    let (four,) = try_indices!(slice, 4)?;
    Ok(four)
}

fn two() -> Result<(&'static mut i32, &'static mut i32), indices::TryIndicesError> {
    let mut data = [5, 4, 3, 2, 1];
    let slice = data.as_mut_slice();
    let (one, four) = try_indices!(slice, 1, 4)?;
    Ok((one, four))
}

fn three() -> Result<&'static mut i32, indices::TryIndicesError> {
    let mut data = [5, 4, 3, 2, 1];
    let slice = data.as_mut_slice();
    let (one, _two, _three) = try_indices!(slice, 1, 2, 3)?;
    Ok(one)
}

fn four() -> Result<(&'static mut i32, &'static mut i32, &'static mut i32, &'static mut i32), indices::TryIndicesError> {
    let mut data = [5, 4, 3, 2, 1];
    let slice = data.as_mut_slice();
    let (one, two, three, four) = try_indices!(slice, 1, 2, 3, 4)?;
    Ok((one, two, three, four))
}

fn five() -> Result<(&'static mut i32, &'static mut i32, &'static mut i32, &'static mut i32, &'static mut i32), indices::TryIndicesError> {
    let mut data = [5, 4, 3, 2, 1];
    let slice = data.as_mut_slice();
    let (zero, one, two, three, four) = try_indices!(slice, 0, 1, 2, 3, 4)?;
    Ok((zero, one, two, three, four))
}
