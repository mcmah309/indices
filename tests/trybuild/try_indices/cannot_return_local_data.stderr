error[E0515]: cannot return value referencing local variable `data`
  --> tests/trybuild/try_indices/cannot_return_local_data.rs:15:5
   |
13 |     let slice = data.as_mut_slice();
   |                 ---- `data` is borrowed here
14 |     let (four,) = try_indices!(slice, 4)?;
15 |     Ok(four)
   |     ^^^^^^^^ returns a value referencing data owned by the current function

error[E0515]: cannot return value referencing local variable `data`
  --> tests/trybuild/try_indices/cannot_return_local_data.rs:22:5
   |
20 |     let slice = data.as_mut_slice();
   |                 ---- `data` is borrowed here
21 |     let (one, four) = try_indices!(slice, 1, 4)?;
22 |     Ok((one, four))
   |     ^^^^^^^^^^^^^^^ returns a value referencing data owned by the current function

error[E0515]: cannot return value referencing local variable `data`
  --> tests/trybuild/try_indices/cannot_return_local_data.rs:29:5
   |
27 |     let slice = data.as_mut_slice();
   |                 ---- `data` is borrowed here
28 |     let (one, _two, _three) = try_indices!(slice, 1, 2, 3)?;
29 |     Ok(one)
   |     ^^^^^^^ returns a value referencing data owned by the current function

error[E0515]: cannot return value referencing local variable `data`
  --> tests/trybuild/try_indices/cannot_return_local_data.rs:36:5
   |
34 |     let slice = data.as_mut_slice();
   |                 ---- `data` is borrowed here
35 |     let (one, two, three, four) = try_indices!(slice, 1, 2, 3, 4)?;
36 |     Ok((one, two, three, four))
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^ returns a value referencing data owned by the current function

error[E0515]: cannot return value referencing local variable `data`
  --> tests/trybuild/try_indices/cannot_return_local_data.rs:43:5
   |
41 |     let slice = data.as_mut_slice();
   |                 ---- `data` is borrowed here
42 |     let (zero, one, two, three, four) = try_indices!(slice, 0, 1, 2, 3, 4)?;
43 |     Ok((zero, one, two, three, four))
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ returns a value referencing data owned by the current function
